src/widget/wtrackmenu.cpp

```diff
@@ -2140,6 +2168,12 @@ bool WTrackMenu::featureIsEnabled(Feature flag) const {
                         TrackModel::Capability::RemoveCrate);
     case Feature::Metadata:
         return m_pTrackModel->hasCapabilities(TrackModel::Capability::EditMetadata);
+    case Feature::Reanalyze:
+        // TODO: Do we actually need the EditMetadata capability here?
```

Comment:
When analyzing tracks their metadata needs to be updated for storing the results. Clearing the beat grid beforehand doesn't really matter imho.

`TrackModel::Capability::Analyze` should already imply `TrackModel::Capability::EditMetadata`. These capability flags are inconsistent and overlapping in their meaning. Everyone just extended it to keep it working. Unrelated, has to be resolved separately.

Reply:
> Clearing the beat grid beforehand doesn't really matter imho.

Yes, it does. Otherwise the beats where not reanalyzed inmy tests.